src/utils/object.ts

```diff
@@ -235,7 +245,18 @@ export function getAllOwnKeys(object: object): IterableIterator<string | symbol>
 
 /**
  * Returns all own entries of an object, including non-enumerable ones and symbols.
+ *
+ * @param obj
  */
-export function getAllOwnEntries<T>(object: { [s: PropertyKey]: T }): IterableIterator<[key: string | symbol, value: T]> {
-  return map(getAllOwnKeys(object), key => [key, object[key]]);
+export function getAllOwnEntries<T>(obj: { [s: PropertyKey]: T }): IterableIterator<[key: string | symbol, value: T]>;
+export function getAllOwnEntries(obj: object): IterableIterator<[key: string | symbol, value: unknown]>;
```

Comment:
```suggestion
```

Reply:
that overload is necessary for objects that aren't simple records, such as interfaces

Reply:
Sorry, I didn't add a comment. It seems you do the same overload twice